backend/search/models.py

```diff
@@ -59,18 +59,41 @@ def _get_cache_path(self) -> pathlib.Path:
         settings.CACHING_DIR.mkdir(exist_ok=True, parents=True)
         return settings.CACHING_DIR / str(self.id)
 
-    def check_results(self) -> bool:
+    def check_cache(self) -> bool:
+        """Check if the cache file is readable and if its contents corresponds
+        to what the database knows about it. If not, the search will have to 
+        be repeated. If search is not yet completed, the cache size is not
+        tested."""
+        cache_path = self._get_cache_path()
         try:
-            self.get_results()
+            with cache_path.open():
+                pass
+        except OSError:
+            return False
+        # Cache file is readable, now check if size is correct
+        if self.search_completed is None:
             return True
-        except Exception:
-            logger.exception('Failed reading results of ComponentSearchQuery: %d', self.pk)
-
-        return False
+        else:
+            actual_size = cache_path.stat().st_size
+            size_equal = actual_size == self.cache_size
+            if not size_equal:
+                logger.warning(
+                    f"Cache size for {self} is {actual_size} while it is supposed to be {self.cache_size}"
+                )
+            return size_equal
```

Comment:
So let's assume there's a cache file with the wrong size (for whatever reason) what happens next?
would it be regenerated on the next query? do we have to manually remove it?